src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs

```diff
@@ -126,23 +115,17 @@ internal bool IsXmlResolverSet
         }
 
         //This is used by get XmlResolver in Xsd.
-        //Check if the config set to prohibit default resovler
+        //Check if the config set to prohibit default resolver
         //notice we must keep GetXmlResolver() to avoid dead lock when init System.Config.ConfigurationManager
         internal XmlResolver? GetXmlResolver_CheckConfig()
         {
-            if (!LocalAppContextSwitches.AllowDefaultResolver && !IsXmlResolverSet)
-                return null;
-            else
-                return _xmlResolver;
+            return LocalAppContextSwitches.AllowDefaultResolver || IsXmlResolverSet ? _xmlResolver : null;
         }
 
         // Text settings
         public int LineNumberOffset
         {
-            get
-            {
-                return _lineNumberOffset;
-            }
+            get => _lineNumberOffset;
             set
             {
                 CheckReadOnly(nameof(LineNumberOffset));
```

Comment:
Depending on what the parameter for CheckReadOnly is used for (is is public at any point?) it might be possible to change the function use use `[CallerMemberName]`